packages/nx/src/generators/utils/project-configuration.ts

```diff
@@ -478,13 +477,13 @@ function readRawWorkspaceJson(tree: Tree): RawProjectsConfigurations {
       (file) => readJson(tree, file)
     ).projects;
     // We already have built a cache
-    if (!staticFSWorkspace) {
-      staticFSWorkspace = buildWorkspaceConfigurationFromGlobs(
-        nxJson,
-        [...globForProjectFiles(tree.root, nxJson)],
-        (file) => readJson(tree, file)
-      );
-    }
+    // if (!staticFSWorkspace) {
+    const staticFSWorkspace = buildWorkspaceConfigurationFromGlobs(
+      nxJson,
+      [...globForProjectFiles(tree.root, nxJson)],
+      (file) => readJson(tree, file)
+    );
+    // }
```

Comment:
We can't just remove this imho, this function gets called frequently when a generator calls `readProjectConfiguration` and the glob op is .5 second each call.

Reply:
If I revert that bunch of tests fails.

Could this mean that those tests were faulty?

Reply:
The problem occurs because the workspace gets cached between test runs.
So depending on the order of invocation `projects.size` has a different value.

Reply:
Since this discussion was resolved, merging it in